src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ExclusionsHelper.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+internal class ExclusionsHelper
+{
+    private const string NullSuffix = "NULL_SUFFIX";
+
+    private string _exclusionsFileName = string.Empty;
+
+    // Use this to narrow down the scope of exclusions to a specific category.
+    // For instance, setting this to "test-templates" will consider 
+    // "src/test-templates/exclusions.txt" but not "src/arcade/exclusions.txt".
+    private Regex? _exclusionRegex;
+
+    private Dictionary<string, HashSet<string>> _suffixToExclusions;
+
+    private Dictionary<string, HashSet<string>> _suffixToUnusedExclusions;
```

Comment:
These can all have the `readonly` keyword since they get set in the constructor and not again after that.